core/chaincode/exectransaction_test.go

```diff
@@ -192,6 +190,28 @@ func TestExecuteDeployTransaction(t *testing.T) {
 	closeListenerAndSleep(lis)
 }
 
+// Test deploy of a transaction
+func TestExecuteDeployTransaction(t *testing.T) {
+	executeDeployTransaction(t, "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example01")
+}
+
+// Test deploy of a transaction with a GOPATH with multiple elements
+func TestGopathExecuteDeployTransaction(t *testing.T) {
+	// add a trailing slash to GOPATH
+	// and a couple of elements - it doesn't matter what they are
+	os.Setenv("GOPATH", os.Getenv("GOPATH") + string(os.PathSeparator) + string(os.PathListSeparator) + "/tmp/foo" + string(os.PathListSeparator) + "/tmp/bar")
+	fmt.Printf("set GOPATH to: \"%s\"\n", os.Getenv("GOPATH"))
+	executeDeployTransaction(t, "github.com/hyperledger/fabric/examples/chaincode/go/chaincode_example01")
+}
+
+// Test deploy of a transaction with a chaincode over HTTP.
+func TestHTTPExecuteDeployTransaction(t *testing.T) {
+	// The chaincode used here cannot be from the hyperledger repo
+	// itself or it won't be downloaded because it will be found
+	// in GOPATH, which would defeat the test
+	executeDeployTransaction(t, "http://github.com/lehors/fabric/examples/chaincode/go/chaincode_example01")
```

Comment:
Ok, that seems reasonable.  Is it possible to at least use a versioned URL rather than leaving it wide open?

E.g. https://github.com/lehors/fabric/tree/ca8135b1a0f95127c96478518628a65a377ec123/examples/chaincode/go/chaincode_example01

Or is there a reason this is a bad idea?